File: 프로그래머스/1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.py
def solution(data, ext, val_ext, sort_by):
    answer = []
    if sort_by == "code": i = 0
    elif sort_by == "date":i=1
    elif sort_by == "maximum":i=2
    elif sort_by == "remain":i=3
    
    if ext == "code": i2 = 0
    elif ext == "date":i2=1
    elif ext == "maximum":i2=2
    elif ext == "remain":i2=3
    sort = sorted(data,key=lambda x:x[i])
    for n in sort:
        a,b,c,d = n
        if i2==0:
            if a < val_ext:
                answer.append(n)
        if i2==1:
            if b < val_ext:
                answer.append(n)
        if i2==2:
            if c < val_ext:
                answer.append(n)
        if i2==3:
            if d < val_ext:
                answer.append(n)
    return answer

File: 프로그래머스/1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<vector<int>> solution(vector<vector<int>> data, string ext, int val_ext, string sort_by) {
    vector<vector<int>> answer;
    vector<vector<int>> res;
    vector<string> std = {"code", "date","maximum","remain"};
    int idx_ext = find(std.begin(), std.end(), ext) - std.begin();
    int idx_sort = find(std.begin(), std.end(), sort_by) - std.begin();
    for(int i = 0; i < data.size(); i++){
        if(data[i][idx_ext] <= val_ext){
            res.push_back(data[i]);
        }
    }
    sort(res.begin(), res.end(), [idx_sort](vector<int> &a, vector<int> &b){return a[idx_sort] < b[idx_sort];});
    return res;
}